sgkit_vcf/utils.py

```diff
@@ -95,3 +100,52 @@ def open_gzip(path: PathType, storage_options: Optional[Dict[str, str]]) -> IO[A
     storage_options = storage_options or {}
     openfile: IO[Any] = fsspec.open(url, compression="gzip", **storage_options)
     return openfile
+
+
+@contextmanager
+def temporary_directory(
+    suffix: Optional[str] = None,
+    prefix: Optional[str] = None,
+    dir: Optional[PathType] = None,
+    storage_options: Optional[Dict[str, str]] = None,
+) -> Generator[str, None, None]:
+    """Create a temporary directory in a fsspec filesystem.
+
+    Parameters
+    ----------
+    suffix : Optional[str], optional
+        If not None, the name of the temporary directory will end with that suffix.
+    prefix : Optional[str], optional
+        If not None, the name of the temporary directory will start with that prefix.
+    dir : Optional[PathType], optional
+        If not None, the temporary directory will be created in that directory, otherwise
+        the local filesystem directory returned by `tempfile.gettempdir()` will be used.
+        The directory may be specified as any fsspec URL.
+    storage_options : Optional[Dict[str, str]], optional
+        Any additional parameters for the storage backend (see `fsspec.open`).
+
+    Yields
+    -------
+    Generator[str, None, None]
+        A context manager yielding the fsspec URL to the created directory.
+    """
+
+    # Fill in defaults
+    suffix = suffix or ""
+    prefix = prefix or ""
+    dir = dir or tempfile.gettempdir()
+    storage_options = storage_options or {}
+
+    # Find the filesystem by looking at the URL scheme (protocol), empty means local filesystem
+    protocol = urlparse(str(dir)).scheme
+    fs = fsspec.filesystem(protocol, **storage_options)
+
+    # Construct a random directory name (use Path purely for filename manipulation, not for IO)
+    # The UUID is based on the host ID, current time,and a random component
+    tempdir = str(Path(dir) / (prefix + str(uuid.uuid1()) + suffix))
```

Comment:
I wonder if we're leaking information here unnecessarily (who knows how the hostname/system time could be used), as if it would be better to use uuid4 altogether. Having the hostname in there makes it no more unique in practice (possibly less, actualy), and the proba of a uuid4 collision is officially negligible.